packages/sdk/src/modules/JsonRpcDatasource.ts

```diff
@@ -0,0 +1,240 @@
+import { Transaction as BTCTransaction } from "bitcoinjs-lib";
+import { rpc } from "../api/jsonrpc";
+import { OrditSDKError } from "../errors";
+import { BaseDatasource } from "./BaseDatasource";
+import { DatasourceUtility } from "./DatasourceUtility";
+import type {
+  GetBalanceOptions,
+  GetInscriptionOptions,
+  GetInscriptionsOptions,
+  GetInscriptionUTXOOptions,
+  GetSpendablesOptions,
+  GetTxOptions,
+  GetUnspentsOptions,
+  GetUnspentsResponse,
+  RelayOptions,
+} from "../api/types";
+import type { Network } from "../config/types";
+import type { Transaction, UTXO, UTXOLimited } from "../transactions/types";
+import type { Inscription } from "../inscription/types";
+import type { JsonRpcPagination } from "./types";
+
+interface JsonRpcDatasourceOptions {
+  network: Network;
+}
+
+class JsonRpcDatasource extends BaseDatasource {
+  constructor({ network }: JsonRpcDatasourceOptions) {
+    super({ network });
+  }
+
+  async getBalance({ address }: GetBalanceOptions) {
+    if (!address) {
+      throw new OrditSDKError("Invalid request");
+    }
+
+    return rpc[this.network].call<number>(
```

Comment:
@iamcrazycoder do we have a mock success/failure response? any specific temp address i could use to test this?

Reply:
Any address should work 👀 

[Link](https://sado.space/docs/api-address#:~:text=to%20an%20address.-,Get%20Balance,-Get%20the%20total) to the doc